apps/dav/lib/Files/Sharing/FilesDropPlugin.php

```diff
@@ -64,7 +64,7 @@ public function beforeMethod(RequestInterface $request, ResponseInterface $respo
 		// Extract the attributes for the file request
 		$isFileRequest = false;
 		$attributes = $this->share->getAttributes();
-		$nickName = $request->getHeader('X-NC-Nickname');
+		$nickName = $request->hasHeader('X-NC-Nickname') ? urldecode($request->getHeader('X-NC-Nickname')) : null;
```

Comment:
Why not simply:

```suggestion
		$nickName = urldecode($request->getHeader('X-NC-Nickname'));
```

if there is no header then the empty string should be returned, this would simplify the condition below (line 73):

```diff
- if ($isFileRequest && ($nickName == null || trim($nickName) === '')) {
+ if ($isFileRequest && trim($nickName) === '') {
			throw new MethodNotAllowed('Nickname is required for file requests');
}
```

Reply:
Because `getHeader` can return null and then `urldecode` throws